mllib/src/main/scala/org/apache/spark/mllib/linalg/distributed/RowMatrix.scala

```diff
@@ -775,6 +778,30 @@ class RowMatrix @Since("1.0.0") (
         s"The number of rows $m is different from what specified or previously computed: ${nRows}.")
     }
   }
+
+  /**
+   * Computing desired tree aggregate depth necessary to avoid exceeding
+   * driver.MaxResultSize during aggregation.
+   * Based on the formulae: (numPartitions)^(1/depth) * objectSize <= DriverMaxResultSize
+   * @param aggregatedObjectSizeInMb the size, in megabytes, of the object being tree aggregated
+   */
+  private[spark] def getTreeAggregateIdealDepth(aggregatedObjectSizeInMb: Int) = {
+    val maxDriverResultSizeInMb = rows.conf.get[Long](MAX_RESULT_SIZE) / (1024 * 1024)
+    logWarning(
+      s"object size to aggregate ($aggregatedObjectSizeInMb MB) exceeds max driver result size"
```

Comment:
Now this always logs a warning? Looks like this should be removed

Reply:
Indeed, done.